### DISCUSSION TITLE
Where userstyles are stored locally on the computer?

### DISCUSSION THREAD

**Discussion by TuTAH1** [openstyles/stylus#1558] [Q&A]
I reinstalled browser and somewhy all userstyles are gone. Last userstyles backup was made a whole half year ago. But! Before reinstalling the browser, I made a backup of a whole browser folder (including User Data), so how can I get my userstyles from a backup?

**Comment by tophf** [comment]
Exit the browser, create `IndexedDB` in the new profile and copy `clngdbkpkpeebahjckkjfobafhncgmne` (this is an id of Stylus extension) from the old profile's `IndexedDB`. Do the same for `Local Extension Settings` and `Sync Extension Settings`.

**Reply by TuTAH1** [reply]
> Exit the browser, create `IndexedDB` in the new profile and copy `clngdbkpkpeebahjckkjfobafhncgmne` (this is an id of Stylus extension) from the old profile's `IndexedDB`. Do the same for `Local Extension Settings` and `Sync Extension Settings`.

I tryed to do that, but styles just cleared (as they where after reinstall. I wasn't cleared Browser's AppData when reinstalling)

**Comment by TuTAH1** [comment]
Can I extract pure css from the backup?

**Reply by eight04** [reply]
If you can launch the browser (downgrade maybe?) from the backup profile then it should be pretty easy to export those data. Otherwise you may have to try finding the database and extract the data using some kind of database management software.
